File: FinalProject/JavaScript/ObjectiveOriented/Controller.js


const LowerCaser = require("./LowerCaser");
const NonEnglishFilter= require("./NonEnglishFilter");
const Splitter= require("./Splitter");
const UniqueWordCounter=require("./UniqueWordCounter");
const LineFilter = require("./LineFilter");
const DocumentProcessorController = require("./DocumentProcessorController");
class Controller {}

let processorController = new DocumentProcessorController();

processorController.beginProcess(process.argv);

module.exports = Controller;

//TODO TEXT
//
// const fs = require('fs');
// const fileUrl = new URL("file:C:\\Users\\dgmon\\YU\\test.txt");
// let text = fs.readFileSync(fileUrl).toString();
//
// let lineFilter = new LineFilter("this");
// ans = lineFilter.process(text);

// let lowerCaser = new LowerCaser();
// ans = lowerCaser.process(ans);
//
// let splitter = new Splitter();
// ans = splitter.process(ans);
//
// let nonEnglishFilter = new NonEnglishFilter();
// ans = nonEnglishFilter.process(ans);
//
// let uniqueWordCounter = new UniqueWordCounter();
// ans = uniqueWordCounter.process(ans);
//
//console.log(ans);
// for(i in ans){
//     console.log(`${i}, ${ans[i]}`);
// }
//

File: FinalProject/Java/src/OOprogram/Controller.java
package OOprogram;

import java.io.IOException;
import java.util.List;
import java.util.Map;
import java.util.Set;

public class Controller {
    public static void main(String args[] ) throws IOException {
        DocumentProcessorController processor = new DocumentProcessorController();
        processor.beginProcess(args);

        //TODO GROUP TESTING
//        DocumentProcessorController processor = new DocumentProcessorController();
//        String[] wc = {"wc","C:\\Users\\dgmon\\YU\\test.txt"};
//        String[] grep = {"grep","this", "C:\\Users\\dgmon\\YU\\test.txt"};
//        String[] grepWc = {"grep","this", "C:\\Users\\dgmon\\YU\\test.txt", "|", "wc"};
//
//        processor.beginProcess(wc);
//        processor.beginProcess(grep);
//        processor.beginProcess(grepWc);
        //TODO INDIVIDUAL TESTING
//        String answer;
//        List<String> listAnswer;
//        Set<Map.Entry> setAnswer;
//        String path = ("C:\\Users\\dgmon\\YU\\test.txt");
//        String text = DocumentProcessorController.pathToString(path);
//        System.out.println(text);
//        //TODO TEXTFILTER
//        TextFilter filter = new TextFilter("this");
//        //System.out.println(filter.process(text));
//        //TODO CONVERT CASE
//        LowerCaser caser = new LowerCaser();
//        answer = caser.process(text);
//        System.out.println(answer);
//
//        //TODO FIND WORDS
//        Splitter splitter = new Splitter();
//        listAnswer = splitter.process(answer);
//        System.out.println(listAnswer);
//
//        //TODO REMOVE NONABC
//        NonABCFilter abcFilter = new NonEnglishFilter();
//        listAnswer = abcFilter.process(listAnswer);
//        System.out.println(listAnswer);
//
//         //TODO COUNT WORDS
//        UniqueWordCounter wordCounter = new UniqueWordCounter();
//        setAnswer = wordCounter.process(listAnswer);
//        System.out.println(setAnswer);

    }
}
